release_tools/changelog.py

```diff
@@ -0,0 +1,329 @@
+
+The tool provides an interactive mode to help the user to
+define the relevant fields.
+"""
+
+import enum
+import os
+import subprocess
+
+import click
+import yaml
+
+
+YAML_FILE_EXTENSION = '.yml'
+
+# GNU tar has a 99 character limit
+MAX_FILENAME_LENGTH = 99 - len(YAML_FILE_EXTENSION)
+
+
+class GitHandler:
+    """Class to help to run Git commands."""
+
+    def __init__(self):
+        self.gitenv = {
+            'LANG': 'C',
+            'PAGER': '',
+            'HTTP_PROXY': os.getenv('HTTP_PROXY', ''),
+            'HTTPS_PROXY': os.getenv('HTTPS_PROXY', ''),
+            'NO_PROXY': os.getenv('NO_PROXY', ''),
+            'HOME': os.getenv('HOME', '')
+        }
+
+    @property
+    def root_path(self):
+        path_ = self._get_submodule_root_path()
+
+        if not path_:
+            basepath = self._get_root_path()
+
+        return basepath
+
+    def _get_root_path(self):
+        cmd = ['git', 'rev-parse', '--show-toplevel']
+        root_path = self._exec(cmd, env=self.gitenv).strip('\n')
+        return root_path
+
+    def _get_submodule_root_path(self):
+        cmd = ['git', 'rev-parse', '--show-superproject-working-tree']
+        root_path = self._exec(cmd, env=self.gitenv).strip('\n')
+        return root_path
+
+    @staticmethod
+    def _exec(cmd, cwd=None, env=None):
+        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE,
+                                cwd=cwd, env=env)
+        (outs, errs) = proc.communicate()
+        return outs.decode('utf-8', errors='surrogateescape')
+
+
+@enum.unique
+class CategoryChange(enum.Enum):
+    """Possible category types."""
+
+    def __new__(cls, value, category, title):
+        obj = object.__new__(cls)
+        obj._value_ = value
+        obj.category = category
+        obj.title = title
+        return obj
+
+    ADDED = (1, 'added', 'New feature')
+    FIXED = (2, 'fixed', 'Bug fix')
+    CHANGED = (3, 'changed', 'Feature change')
+    DEPRECATED = (4, 'deprecated', 'New deprecation')
+    REMOVED = (5, 'removed', 'Feature removal')
+    SECURITY = (6, 'security', 'Security fix')
+    PERFORMANCE = (7, 'performance', 'Performance improvement')
+    OTHER = (8, 'other', 'Other')
+
+    @classmethod
+    def values(cls):
+        return [member.category for member in cls]
+
+
+class ChangelogEntry:
+    """Class to store changelog entries data."""
+
+    def __init__(self, title, category, author, pr=None, notes=None):
+        self.title = title
+        self.category = category
+        self.author = author
+        self.pr = pr
+        self.notes = notes
+
+    def to_dict(self):
+        return {
+            'title': self.title,
+            'category': self.category,
+            'author': self.author,
+            'pull_request': self.pr,
+            'notes': self.notes
+        }
+
+
+def title_prompt():
+    prompt_msg = ">> Please specify the title of your change"
```

Comment:
Please add a small description of the method

Reply:
Added.